lib/agent0/agent0/hyperdrive/interactive/i_hyperdrive.py

```diff
@@ -0,0 +1,532 @@
+"""Defines the interactive hyperdrive class that encapsulates a hyperdrive pool."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import os
+from dataclasses import asdict, dataclass
+from typing import Any, Literal, Type, overload
+
+import nest_asyncio
+import numpy as np
+from eth_account.account import Account
+from eth_account.signers.local import LocalAccount
+from ethpy import EthConfig
+from ethpy.base import set_anvil_account_balance, smart_contract_transact
+from ethpy.hyperdrive import (
+    HyperdriveAddresses,
+    HyperdriveReadWriteInterface,
+    ReceiptBreakdown,
+    fetch_hyperdrive_address_from_uri,
+)
+from fixedpointmath import FixedPoint
+from numpy.random._generator import Generator
+from web3 import Web3
+
+from agent0.hyperdrive import HyperdriveActionType, HyperdriveAgent, TradeResult, TradeStatus
+from agent0.hyperdrive.agent import build_wallet_positions_from_chain
+from agent0.hyperdrive.crash_report import log_hyperdrive_crash_report
+from agent0.hyperdrive.exec import async_execute_agent_trades, set_max_approval
+from agent0.hyperdrive.policies import HyperdriveBasePolicy
+from agent0.test_utils import assert_never
+
+from .event_types import (
+    AddLiquidity,
+    CloseLong,
+    CloseShort,
+    OpenLong,
+    OpenShort,
+    RedeemWithdrawalShares,
+    RemoveLiquidity,
+)
+from .i_chain import IChain
+from .i_hyperdrive_agent import IHyperdriveAgent
+from .i_hyperdrive_policy import IHyperdrivePolicy
+
+# In order to support both scripts and jupyter notebooks with underlying async functions,
+# we use the nest_asyncio package so that we can execute asyncio.run within a running event loop.
+nest_asyncio.apply()
+
+
+class IHyperdrive:
+    """Interactive Hyperdrive class that supports connecting to an existing hyperdrive deployment."""
+
+    @dataclass(kw_only=True)
+    class Config:
+        """The configuration for the interactive hyperdrive class
```

Comment:
pls add a period at the end of the top-line desc & a newline between the top-line desc and the Attributes